src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupFreePeriodResource.java

```diff
@@ -177,11 +187,19 @@ public ResponseEntity<TutorialGroupFreePeriod> create(@PathVariable Long courseI
     public ResponseEntity<Void> delete(@PathVariable Long courseId, @PathVariable Long tutorialGroupsConfigurationId, @PathVariable Long tutorialGroupFreePeriodId)
             throws URISyntaxException {
         log.debug("REST request to delete TutorialGroupFreePeriod: {}", tutorialGroupFreePeriodId);
+        log.info("REST request to delete TutorialGroupFreePeriod: {} of tutorial group configuration {} of course: {}", tutorialGroupFreePeriodId, tutorialGroupsConfigurationId,
+                courseId);
         TutorialGroupFreePeriod tutorialGroupFreePeriod = tutorialGroupFreePeriodRepository.findByIdElseThrow(tutorialGroupFreePeriodId);
         checkEntityIdMatchesPathIds(tutorialGroupFreePeriod, Optional.ofNullable(courseId), Optional.ofNullable(tutorialGroupsConfigurationId));
         authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFreePeriod.getTutorialGroupsConfiguration().getCourse(), null);
-        tutorialGroupFreePeriodService.activateOverlappingSessions(tutorialGroupFreePeriod);
-        tutorialGroupFreePeriodRepository.delete(tutorialGroupFreePeriod);
+        Optional<TutorialGroupsConfiguration> configurationOptional = tutorialGroupsConfigurationRepository.findByIdWithEagerTutorialGroupFreePeriods(courseId);
```

Comment:
This Optional does not seem to contain a value when trying to delete a FreePeriod